kstreams/middleware/middleware.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import typing
+
+from aiokafka import errors
+
+from kstreams import ConsumerRecord, types
+
+if typing.TYPE_CHECKING:
+    from kstreams import Stream  #  pragma: no cover
+
+
+logger = logging.getLogger(__name__)
+
+
+class Middleware(typing.Protocol):
```

Comment:
What about using [ParamSpec](https://github.com/encode/starlette/blob/master/starlette/middleware/__init__.py#L11C5-L14) like in starlette?

Reply:
We need to update the `mypy` version in order to try it (we are using a really old version). Next release we can do it